File: Python/섹션 3 - (진행중) 탐색&시뮬레이션(string,1차원,2차원 리스트 탐색)/6. 격자판 최대합 (최상)/AA.py
import sys
#sys.stdin=open('input.txt', 'rt')

num=int(input())
arr=[list(map(int,input().split())) for _ in range(num)]
# arr=[list(map(int,input().split()))] * num

# 테스트 출력용
# for x in arr:
#     print(x)

max=0
# 가로 합 (O(n))
# 세로 합 (O(n))
# -> 같이 진행되므로, O(2n^2)이 아닌, (n^2)의 복잡도
for i in range(num):
    sum1=sum2=0 # 한번에 초기화
    for j in range(num):
        sum1+=arr[i][j]
        sum2+=arr[j][i]
    if max<sum1:
        max=sum1
    if max<sum2:
        max=sum2

# 대각선 합 (O(2) - 상수 복잡도)
sum1=0
sum2=0
for i in range(num):
    sum1+=arr[i][i]
    sum2+=arr[num-i-1][i]
if max<sum1:
    max=sum1
if max<sum2:
    max=sum2

print(max)

File: Javascript/섹션2. (진행중) 1-2차원 배열 탐색/6. 격자판 최대합 (상)/AA.js
function solution(num, arr){
    
    let sum1=0
    let sum2=0
    let max=0

    // 수직,수평의 합
    for(let i=0;i<n;i++){
        for(let j=0;j<n;j++){
            sum1+=arr[i][j]
            sum2+=arr[i][j]
        }
        if(sum1>max) max=sum1
        if(sum2>max) max=sum2
        sum1=0
        sum2=0
    }

    // 대각선의 합
    for(let i=0;i<n;i++){
        sum1+=arr[i][i]
        sum2+=arr[i][n-i-1]
        if(sum1>max) max=sum1
        if(sum2>max) max=sum2
    }

    return max
}
let arr=[]
let n=5

arr.push([10, 13, 10, 12, 15])
arr.push([12, 39, 30, 23, 11])
arr.push([11, 25, 50, 53, 15])
arr.push([19, 27, 29, 37, 27])
arr.push([19, 13, 30, 13, 19])

console.log(solution(n, arr))